qutebrowser/utils/jinja.py

```diff
@@ -127,3 +130,33 @@ def render(template, **kwargs):
 
 environment = Environment()
 js_environment = jinja2.Environment(loader=Loader('javascript'))
+
+
+@functools.lru_cache()
+def template_config_variables(template: str) -> typing.FrozenSet[str]:
+    """Return a frozenset of config variables that is used in template."""
+    pending = [environment.parse(template)]  # a list of unvisited nodes
+    result = []  # type: typing.List[str]
+    while pending:
+        node = pending.pop()
+        if isinstance(node, jinja2.nodes.Getattr):
+            # List of attribute names in reverse order.
+            # For example it's ['ab', 'c', 'd'] for 'conf.d.c.ab'.
+            attrlist = []  # type: typing.List[str]
+            while True:
```

Comment:
```suggestion
            while isinstance(node, jinja2.nodes.Getattr):
```

makes the while loop clearer, and you can skip the `break` below.

Reply:
I don't think there's any difference about code readability, but your suggested change adds an is_instance check. Not that this is performance critical code, but no reason to sacrifice performance.

Reply:
I disagree - I find it much more readable to have the condition in the `while ...:` part (rather than have a `while True:` with a `break`. There's also no reason to sacrifice readability if there isn't an actual performance difference :wink:

Reply:
Blame Python for not having a `do .. while` or `repeat .. until` loop.